frontend/packages/console-shared/src/components/synced-editor/index.tsx

```diff
@@ -59,6 +63,7 @@ export const SyncedEditor: React.FC<SyncedEditorProps> = ({
       .then((js) => {
         setSwitchError(undefined);
         handleFormDataChange(js);
+        setYAML('');
```

Comment:
Can you help me understand this change? I can't see any difference between having it and not having it.

Please provide steps to reproduce whatever the issue is to show that it is still needed. If it is needed, I suspect there is other logic flow issues afloat. But I can't debug that until I can see this issue pop up.

Reply:
> Can you help me understand this change? I can't see any difference between having it and not having it.
> 
> Please provide steps to reproduce whatever the issue is to show that it is still needed. If it is needed, I suspect there is other logic flow issues afloat. But I can't debug that until I can see this issue pop up.

@andrewballantyne Steps to reproduce:
1. Remove the setter `setYAML('')`
2. Click on yaml view from form view.
3. Update `maxUnavailable` or `minAvailable`
4. Save - the previous value is restored
5. Repeat 3 and 4 -  An error occurred

Reply:
Okay, not sure how this hasn't caused more bugs... but this component fails to be a properly uncontrolled component and thus is getting you into this situation. It's only used in two other places (Create Network Policy & Create Operand), so it's possible they are not high trafficked areas and thus the bug was not obvious or common.

The code works fine when using the form and switching to the YAML to inspect the result. But if you edit the YAML directly, and try saving on that page, the content is not properly updated internally and thus saves with the initial data. As far as I can tell, there is no way to force a controlled view of the component either, so this is just a flat bug.

Although I see what you were doing, this is the change you should do:
```suggestion
        setYAML(newYAML);
```
That way you sync up local data with the latest change. TBH... the code should really just set the form and always convert the form to yaml as it is needed or just allow a controlled state... but I don't want to bog your PR down with fixing a component that isn't yours.